packages/client/tests/functional/issues/8832/tests.ts

```diff
@@ -0,0 +1,202 @@
+      await prisma.$transaction(cleanPrismaPromises)
+    }
+
+    describe('issue #8832', () => {
+      beforeEach(async () => {
+        await clean()
+      }, 10_000)
+
+      test('should succeed when "in" has 32766 ids', async () => {
+        const n = 32766
+        const ids = await createTags(n)
+        const tags = await getTagsFindManyIn(ids)
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "include" involves 32766 records', async () => {
+        const n = 32766
+        await createTags(n)
+        const tags = await getTagsFindManyInclude()
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "in" has 32767 ids', async () => {
+        const n = 32767
+        const ids = await createTags(n)
+        const tags = await getTagsFindManyIn(ids)
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "include" involves 32767 records', async () => {
+        const n = 32767
+        await createTags(n)
+        const tags = await getTagsFindManyInclude()
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "in" has 32768 ids', async () => {
+        const n = 32768
+        const ids = await createTags(n)
+        const tags = await getTagsFindManyIn(ids)
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "include" involves 32768 records', async () => {
+        const n = 32768
+        await createTags(n)
+        const tags = await getTagsFindManyInclude()
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "in" has 65536 ids', async () => {
+        const n = 65536
+        const ids = await createTags(n)
+        const tags = await getTagsFindManyIn(ids)
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "include" involves 65536 records', async () => {
+        const n = 65536
+        await createTags(n)
+        const tags = await getTagsFindManyInclude()
+
+        expect(tags.length).toBe(n)
+      })
+
+      /**
+       * Highlight scenarios of the query chunking logic behavior depending on how the bind variables are counted
+       * in the query filters.
+       */
+      describe('unhandled filters', () => {
+        test('should fail with `value too large to transmit` when "in" is repeated at least twice and "n" is $QUERY_BATCH_SIZE', async () => {
+          expect.assertions(3)
+          const n = 32766
+          const ids = await createTags(n)
+
+          try {
+            await prisma.tag.findMany({
+              where: {
+                OR: [
+                  {
+                    id: { in: ids },
+                  },
+                  {
+                    id: { in: ids },
+                  },
+                ],
+              },
+            })
+          } catch (e) {
+            const error = e as Error & { code: number; meta?: unknown }
```

Comment:
maybe
```suggestion
            const error = e as Prisma.PrismaClientKnownRequestError
```
(and import `Prisma` above)

but no strong opinion one way or the other